### PR TITLE
Bt 25.1

### PR DESCRIPTION
**Pull request by peytonshetler**
## Summary

1. I created 3 custom UI components: BaseButton, BaseTextField & BaseViewController
2. I re-created the Landing Screen with these components

#### Characteristics

- [X] Feature completed
- [ ] Experimental
- [X] UI affected (screen shots included)

#### Testing

- [ ] included
- [X] Not Included 

Screen Shot 2020-02-13 at 9 55 06 AM

### CODE REVIEW

**Comment by peytonshetler** [comment]
I should say, I plan on creating a dedicated login screen as well.  That's why there's not nearly as much code in this screen as before.

**Comment by peytonshetler** [comment]
I guess I can put those login/signup buttons in a stackView too, hindsight 👍🏼

File: Been-There/Been-There/View/CustomUI/Button/BaseButton.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupButton()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupButton()
```
Comment (JZDesign) [inline Been-There/Been-There/View/CustomUI/Button/BaseButton.swift]: Is this getting called from your convenience init?

File: Been-There/Been-There/View/CustomUI/Button/BaseButton.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        let fromPoint = CGPoint(x: center.x - 8, y: center.y)
+        let fromValue = NSValue(cgPoint: fromPoint)
+        
+        let toPoint = CGPoint(x: center.x + 8, y: center.y)
+        let toValue = NSValue(cgPoint: toPoint)
+        
+        shake.fromValue = fromValue
+        shake.toValue = toValue
+        
+        layer.add(shake, forKey: "position")
```
Comment (JZDesign) [inline Been-There/Been-There/View/CustomUI/Button/BaseButton.swift]: # 👍

File: Been-There/Been-There/View/CustomUI/TextField/BaseTextField.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    
+    var fontSize: CGFloat = 20
+    
+    convenience init(placeholder: String, withColor placeholderColor: UIColor, textColor: UIColor, hasSecureEntry: Bool? = false) {
+        self.init()
+        
+        attributedPlaceholder = NSAttributedString(string: placeholder, attributes: [
+            NSAttributedString.Key.foregroundColor: placeholderColor,
+            NSAttributedString.Key.font: UIFont(name: "Arial", size: fontSize)!
+        ])
```
Comment (JZDesign) [inline Been-There/Been-There/View/CustomUI/TextField/BaseTextField.swift]: I just learned something. I didn't know about the attributed placeholder.

File: Been-There/Been-There/View/CustomUI/TextField/BaseTextField.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setStyles()
+    }
+    
+    private func setStyles() {
+        //backgroundColor = #colorLiteral(red: 1, green: 1, blue: 1, alpha: 0)
+        backgroundColor = .red
```
Comment (JZDesign) [inline Been-There/Been-There/View/CustomUI/TextField/BaseTextField.swift]: Do you really want the input to be red?

File: Been-There/Been-There/View/CustomUI/Button/BaseButton.swift (hunk 1 above)
Comment (peytonshetler) [reply Been-There/Been-There/View/CustomUI/Button/BaseButton.swift]: The setupButton() function? Maybe it is?  I kind of just assumed it was being taken care of in `self.init()` but maybe that's wrong.  There's so many initializers haha

Convenience init: 
`convenience init(title: String, withColor titleColor: UIColor, bgColor: UIColor) {
        self.init()
        setTitle(title, for: .normal)
        setTitleColor(titleColor, for: .normal)
        backgroundColor = bgColor
    }`

Comment (JZDesign) [reply Been-There/Been-There/View/CustomUI/Button/BaseButton.swift]: I think you can remove the 2 inits and only keep the convenience one.

File: Been-There/Been-There/View/CustomUI/TextField/BaseTextField.swift (hunk 4 above)
Comment (peytonshetler) [reply Been-There/Been-There/View/CustomUI/TextField/BaseTextField.swift]: Oh no, haha.  I'm still working on styling/placement for that!  I left it red b/c I'm gonna use it in the next view controller

**Pull request merged**